pymongo/encryption_options.py

```diff
@@ -58,6 +58,7 @@ def __init__(
         bypass_query_analysis: bool = False,
         encrypted_fields_map: Optional[Mapping[str, Any]] = None,
         key_expiration_ms: Optional[int] = None,
+        is_sync: bool = True,
```

Comment:
This is a public api so we can't add "is_sync" here. What's the motivation for this?

Reply:
Oh, I added `is_sync` as a param to a few functions because in `ssl_support.get_ssl_context()`, we'll need to know which version of ssl should be used,, is there another way to go about this that is preferred?

Reply:
Thanks, we still can not add "is_sync" here so we need to find another way.

Reply:
One way would be to lazily init the async SSLContext in kms_request().

Reply:
I think I did it? Not sure if this is what you meant, but what I have now would leave the params to AutoEncryptOpts unchanged. 
I basically delayed the parse_kms_tls_options because that's where is_sync was needed. I believe i appropriately delayed the definition of kms_tls_options but let me know if I missed something. I'm not super familiar with how encryption in the driver works >.<